crates/stream-download/src/storage/adaptive.rs

```diff
@@ -1,132 +1,186 @@
-//! Storage wrappers that use [bounded](super::bounded) implementations when the content length is
-//! not known. In this scenario, it is assumed that the source is an infinite
-//! stream.
+//! Storage wrappers that adaptively choose between two storage providers based on content length.
 //!
-//! This module is useful when you want to need to support both infinite and finite streams while
-//! avoiding the overhead of using a bounded buffer when it's not necessary.
+//! This module provides a way to use different storage strategies depending on the stream's
+//! characteristics:
+//! - For infinite streams: uses the bounded storage provider
```

Comment:
The terminology gets a little muddy here since we're calling these bounded/unbounded, but we don't actually need to create a `BoundedStorageProvider` instance for the second case. Maybe it would be better to call these fixed/variable instead of bounded/unbounded.

- for infinite streams: uses the fixed-length storage type wrapped in a `BoundedStorageProvider`
- for inite streams smaller than the buffer size: uses the fixed-length storage type
- for finite streams larger than the buffer size: uses the variable-length storage type

Reply:
Yes, good idea.